components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add.jsp

```diff
@@ -153,34 +154,43 @@
-                        if (callbackUrl.startsWith("regexp=")) {
-                            // skip validation
-                        } else if (!isWhiteListed(callbackUrl, ["url"])) {
-                            CARBON.showWarningDialog('<fmt:message key="callback.is.not.url"/>');
-                            return false;
-                        }
-                        if (!isWhiteListed(backChannelLogoutUrl, ["url"]) || !isNotBlackListed(backChannelLogoutUrl,
-                                ["uri-unsafe-exists"])) {
-                            CARBON.showWarningDialog('<fmt:message key="backchannel.logout.is.not.url"/>');
-                            return false;
-                        }
-                    }
                     var value = document.getElementsByName("application")[0].value;
                     if (value == '') {
                         CARBON.showWarningDialog('<fmt:message key="application.is.required"/>');
                         return false;
                     }
+
                     var version2Checked = document.getElementById("oauthVersion20").checked;
                     if (version2Checked) {
                         if (!$(jQuery("#grant_authorization_code"))[0].checked && !$(jQuery("#grant_implicit"))[0].checked) {
                             document.getElementsByName("callback")[0].value = '';
+                        } else {
+                            // This is to support providing regex patterns for callback URLs
+                            if (callbackUrl.startsWith("regexp=")) {
+                                // skip validation
+                            } else if (!(isWhiteListed(callbackUrl, ["https-url"]) || isWhiteListed(callbackUrl, ["http-url"])) || !isNotBlackListed(callbackUrl, ["uri-unsafe-exists"])) {
+                                CARBON.showWarningDialog('<fmt:message key="callback.is.not.url"/>');
+                                return false;
+                            }
+
+                            if (oidcLogoutType === "<%= OAuthConstants.OIDCConfigProperties.BACK_CHANNEL_LOGOUT%>") {
+                                if (!(isWhiteListed(oidcLogoutUrl, ["https-url"]) || isWhiteListed(oidcLogoutUrl, ["http-url"])) || !isNotBlackListed(oidcLogoutUrl,
+                                    ["uri-unsafe-exists"])) {
+                                    CARBON.showWarningDialog('<fmt:message key="logout.is.not.url"/>');
+                                    return false;
+                                }
+                            } else if (oidcLogoutType === "<%= OAuthConstants.OIDCConfigProperties.FRONT_CHANNEL_LOGOUT%>") {
+                                if (!isWhiteListed(oidcLogoutUrl, ["https-url"]) || !isNotBlackListed(oidcLogoutUrl, ["uri-unsafe-exists"])) {
+                                    CARBON.showWarningDialog('<fmt:message key="logout.is.not.https.url"/>');
```

Comment:
Please create an issue to validate the https URL in the backend service as well and address in the next PR